types/videojs-vtt.js/index.d.ts

```diff
@@ -0,0 +1,60 @@
+// Type definitions for videojs-vtt.js 0.15
+// Project: https://github.com/videojs/vtt.js
+// Definitions by: David Ko <https://github.com/davidholyko>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type TimeInSeconds = number;
+type VttText = string;
+type DecoderFn = (data: string) => string;
+interface Decoder {
+    decode: DecoderFn;
+}
+
+export function shim(): void;
+export function restore(): void;
+
+interface VttParser {
+    buffer: string;
+    decoder: Decoder;
+    regionList: string[];
+    state: 'INITIAL' | 'BADCUE' | 'HEADER' | 'NOTE' | 'ID' | 'CUETEXT' | null;
+    vttjs: {};
+    window: Window;
+
+    onregion: (region: VTTRegion) => void;
+    oncue: (cue: VTTCue) => void;
+    onflush: () => void;
+    onparsingerror: (e: Error) => void;
+    parse: (moreData?: VttText) => VttParser;
+    flush: () => void;
+}
+
+interface VttParserConstructor {
+    new (window: Window, stringDecoder?: ReturnType<typeof WebVTT.StringDecoder>): VttParser;
+}
+
+export namespace WebVTT {
+    const Parser: VttParserConstructor;
+    function StringDecoder(): Decoder;
+    function processCues(window: Window, cues: VTTCue[], overlay: HTMLElement | null): void;
+    function convertCueToDOMTree(window: Window, text: string): HTMLElement;
+}
+
+export const VTTCue: {
+    prototype: VTTCue;
+    new (startTime: TimeInSeconds, endTime: TimeInSeconds, text: string): VTTCue;
+
+    // http://www.whatwg.org/specs/web-apps/current-work/multipage/the-video-element.html#text-track-cue-display-state
+    displayState?: any;
+    hasBeenReset?: boolean;
+};
+
+export const VTTRegion: {
+    prototype: VTTRegion;
+    new (): VTTRegion;
+
+    fromJSON(json: any): VTTRegion;
+    create(options: any): VTTRegion;
+};
+
+export {};
```

Comment:
Is this package actually a module? Following through its package.json, `main` is `lib/browser-index.js` (https://unpkg.com/videojs-vtt.js@0.15.5/lib/browser-index.js) which definitely does not contain any exports.

Reply:
yes 😄 

https://github.com/videojs/vtt.js/blob/main/lib/index.js

Reply:
Ugh, my bad. I didn't see the `module.exports` at the top.

Reply:
But, that file is not what is resolved; browser-index is: https://unpkg.com/videojs-vtt.js@0.15.5/lib/browser-index.js

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5341706/60ae04fe-b78b-4527-b532-ba5ddc916963)

But, it works.